### PR TITLE
Add blockthrough placements to existing ad slots.

### PR DESCRIPTION
**Pull request by LATaylor-guardian**
This PR add's span tags of the form `<span class='bt-uid-tg' uid='SOME_ID' style='display: none !important'></span>` within the elements of our existing ad slots so that Blockthrough may target them.

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/dfp/create-slot.js [hunk 1]
```diff
@@ -15,6 +15,17 @@ const inlineDefinition = {
...
 
+const adSlotToBlockthroughUids = {
+    inline1: '5a98587091-157',
+    inline2: '5a98587869-157',
+    inline3: '5a98587f21-157',
+    inline4: '5a985885d5-157',
+    inline5: '5a98588c9c-157',
+    inline6: '5a985892ed-157',
+    inline7: '5a985899d4-157',
+    inline8: '5a9858a114-157',
```
Comment (kelvin-chappell) [inline static/src/javascripts/projects/commercial/modules/dfp/create-slot.js]: I think there's no limit to the number of inline slots.  Could this be done by a prefix comparison instead?

File: common/app/views/fragments/commercial/adSlot.scala.html [hunk 2]
```diff
@@ -6,18 +6,24 @@
     refresh: Boolean = true,
     outOfPage: Boolean = false,
     optId: Option[String] = None,
-    optClassNames: Option[String] = None
+    optClassNames: Option[String] = None,
+    adSlotToBlockthroughUids: Map[String, String] = Map("right" -> "5a9858a84f-157", "top-above-nav" -> "5a98585772-157")
```
Comment (kelvin-chappell) [inline common/app/views/fragments/commercial/adSlot.scala.html]: I wonder if these could be properties of slots defined in the Commercial helper object instead? https://github.com/guardian/frontend/blob/master/common/app/views/support/Commercial.scala

**Review by kelvin-chappell — APPROVED** [review]
I had a couple of suggestions but nothing fatal. 👍

File: static/src/javascripts/projects/commercial/modules/dfp/create-slot.js (hunk 1 above)
Comment (LATaylor-guardian) [reply static/src/javascripts/projects/commercial/modules/dfp/create-slot.js]: The limit is on Blockthrough' side. we Have to provide them with our slots.

File: common/app/views/support/Commercial.scala [hunk 3]
```diff
@@ -23,7 +24,7 @@ object Commercial {
   def shouldShowAds(page: Page): Boolean = page match {
     case c: model.ContentPage if c.item.content.shouldHideAdverts => false
     case p: model.Page if p.metadata.sectionId == "identity" => false
-    case s: model.SimplePage if s.metadata.contentType == "Signup" => false
+    case s: model.SimplePage if s.metadata.contentType.contains(Signup) => false
```
Comment (LATaylor-guardian) [inline common/app/views/support/Commercial.scala]: Noticed this was comparing unrelated types!

File: static/src/javascripts/projects/commercial/modules/dfp/create-slot.js (hunk 1 above)
Comment (kelvin-chappell) [reply static/src/javascripts/projects/commercial/modules/dfp/create-slot.js]: I didn't notice that they were all different!

File: common/app/views/support/Commercial.scala (hunk 3 above)
Comment (hosamaly) [reply common/app/views/support/Commercial.scala]: Can [Scalactic](http://www.scalactic.org/) help avoid such issues? According to the docs:
> A compiler error for an equality comparison that would always yield false looks like:
> ```scala
> import TypeCheckedTripleEquals._
> Some("hi") === "hi"
> error: types Some[String] and String do not adhere to the type constraint
> selected for the === and !== operators; the missing implicit parameter is
> of type org.scalactic.Constraint[Some[String],String]
>           Some("hi") === "hi"

File: common/app/views/fragments/commercial/adSlot.scala.html [hunk 4]
```diff
@@ -14,10 +16,15 @@
...
+    @if(!refresh) { data-refresh="false" }
+    @if(outOfPage) { data-out-of-page="true" }
+    @sizeMapping.map { case(breakpoint, sizes) =>
         data-@breakpoint="@sizes.mkString("|")"
     }
-    >@placeholderContent</div>
+    >@placeholderContent
+
+    @Commercial.getBlockthroughElementUid(name).map { uid =>
+        <span class="bt-uid-tg" uid="@uid" style="display: none !important"></span>
```
Comment (regiskuckaertz) [inline common/app/views/fragments/commercial/adSlot.scala.html]: You can use `<span hidden ...>` instead of using the `style` attribute

Comment (LATaylor-guardian) [reply common/app/views/fragments/commercial/adSlot.scala.html]: This is the tag specified by blockthrough specifically.

**Review by JonNorman — APPROVED** [review]
It looks like this is the best way to go; let's pick @NataliaLKB's brains when she's back to see if there is another alternative.

**Pull request merged**